update-ubuntu-or-debian-server/run.sh

```
@@ -2,193 +2,944 @@
-echo "If this is useful, please consider supporting my work at: https://ko-fi.com/bigbeartechworld"
-echo
-print_decorative_line
+# Function to load configuration
+load_config() {
+    if [ ! -f "$CONFIG_FILE" ]; then
+        mkdir -p "$CONFIG_DIR" 2>/dev/null
+        echo "$DEFAULT_CONFIG" > "$CONFIG_FILE"
+        log "Created default configuration file at $CONFIG_FILE"
+    fi
+    
+    source "$CONFIG_FILE"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)